app/src/main/java/org/simple/clinic/summary/linkId/LinkIdWithPatientEffectHandler.kt

```diff
@@ -46,4 +47,17 @@ class LinkIdWithPatientEffectHandler @AssistedInject constructor(
           .map { IdentifierAddedToPatient }
     }
   }
+
+  private fun getPatientNameFromId(): ObservableTransformer<GetPatientNameFromId, LinkIdWithPatientEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(schedulersProvider.io())
+          .map {
+            PatientNameReceived(patientRepository.patientImmediate(it.patientUuid)!!.fullName
+
+            )
+          }
```

Comment:
We can make this easier to read by using two `.map` operators and a lambda reference instead of a single multi-line `.map {}` block.

```kotlin
.map { patientRepository.patientImmediate(it.patientUuid)!!.fullName }
.map(::PatientNameReceived)
```

You can choose to change this later. For now, I am going to clean up the unnecessary line that was added.